File: Py/Sintaxis II/Tema - Recursividad.py
# ┌────────────────────────
# │  Función recursiva para calcular el factorial
# └────────────────────────
def factorial(x):
    """Esta es una función recursiva
    para encontrar el factorial de un entero"""
    if x == 1:
        return 1
    else:
        return x * factorial(x - 1)


num = 3
print("El factorial de", num, "es", factorial(num))
# Salida: El factorial de 3 es 6

File: Cs/Sintaxis II/Tema - Recursividad.cs
using System;

class Program
{
    // Función recursiva para calcular el factorial
    static int Factorial(int n)
    {
        if (n == 0)
            return 1;
        else
            return n * Factorial(n - 1);
    }

    void ContarAtras(int n)
    {
        if (n == 0)
        {
            Console.WriteLine("¡Despegue!");
            return;
        }

        Console.WriteLine(n);
        ContarAtras(n - 1);
    }

    static void Main(string[] args)
    {
        Console.Write("Ingresa un número: ");
        int num = int.Parse(Console.ReadLine());

        int resultado = Factorial(num);
        Console.WriteLine("El factorial de " + num + " es: " + resultado);

        ContarAtras(5);
    }
}
